ciao-cli/workload.go

```diff
@@ -103,3 +114,186 @@ func (cmd *workloadListCommand) run(args []string) error {
 	}
 	return nil
 }
+
+type workloadCreateCommand struct {
+	Flag     flag.FlagSet
+	yamlFile string
+}
+
+func (cmd *workloadCreateCommand) parseArgs(args []string) []string {
+	cmd.Flag.StringVar(&cmd.yamlFile, "yaml", "", "filename for yaml which describes the workload")
+	cmd.Flag.Usage = func() { cmd.usage() }
+	cmd.Flag.Parse(args)
+	return cmd.Flag.Args()
+}
+
+func (cmd *workloadCreateCommand) usage(...string) {
+	fmt.Fprintf(os.Stderr, `usage: ciao-cli [options] workload create [flags] 
+
+Create a new workload
+
+The create flags are:
+
+`)
+	cmd.Flag.PrintDefaults()
+	os.Exit(2)
+}
+
+func getCiaoWorkloadsResource() (string, error) {
+	return getCiaoResource("workloads", api.WorkloadsV1)
+}
+
+type source struct {
+	Type types.SourceType `json:"service"`
+	ID   string           `json:"id"`
+}
+
+type disk struct {
+	ID       *string `yaml:"volume_id"`
+	Size     int     `yaml:"size"`
+	Bootable bool    `yaml:"bootable"`
+	Source   *source `yaml:"source"`
+}
+
+type defaultResources struct {
+	VCPUs  int `yaml:"vcpus"`
+	MemMB  int `yaml:"mem_mb"`
+	DiskMB int `yaml:"disk_mb"`
+}
+
+// we currently only use the first disk due to lack of support
+// in types.Workload for multiple storage resources.
+type workloadOptions struct {
+	Description     string           `yaml:"description"`
+	VMType          string           `yaml:"vm_type"`
+	FWType          string           `yaml:"fw_type"`
+	ImageName       string           `yaml:"image_name"`
+	ImageID         string           `yaml:"image_id"`
+	Defaults        defaultResources `yaml:"defaults"`
+	CloudConfigFile string           `yaml:"cloud_init"`
+	Disks           []disk           `yaml:"disks"`
+}
+
+func optToReq(opt workloadOptions, req *types.Workload) error {
+	b, err := ioutil.ReadFile(opt.CloudConfigFile)
+	if err != nil {
+		return err
+	}
+
+	config := string(b)
+
+	// this is where you'd validate that the options make
+	// sense.
+	req.Description = opt.Description
+	req.VMType = payloads.Hypervisor(opt.VMType)
+	req.FWType = opt.FWType
+	req.ImageName = opt.ImageName
+	req.ImageID = opt.ImageID
+	req.Config = config
+
+	for _, disk := range opt.Disks {
+		res := types.StorageResource{
+			Size:     disk.Size,
+			Bootable: disk.Bootable,
+		}
```

Comment:
I think it would be good if you could add a Persistent field to disk and populate types.StorageResource from that.